### PR TITLE
[#IOCIT-177 #ICC-276] Add pushnotif routing for canary users

### PR DESCRIPTION
**Pull request by gquadrati**
#### List of Changes

Add canary check using regex

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -924,5 +924,5 @@ export const FF_ROUTING_PUSH_NOTIF_BETA_TESTER_SHA_LIST = pipe(
 export const FF_ROUTING_PUSH_NOTIF_CANARY_SHA_USERS_REGEX = pipe(
   process.env.FF_ROUTING_PUSH_NOTIF_BETA_TESTER_SHA_LIST,
```
Comment (fabriziopapi) [inline src/config.ts:925]: wht using FF_ROUTING_PUSH_NOTIF_BETA_TESTER_SHA_LIST here?

Comment (gquadrati) [reply src/config.ts]: My bad, wrong copy-paste :D Done in [123d52b](https://github.com/pagopa/io-backend/pull/960/commits/123d52bed584e45c391df90c13cb28c3bacf43e7)

File: src/services/__tests__/notificationServiceFactory.test.ts [hunk 2]
```diff
@@ -42,11 +46,12 @@ describe("notificationServiceFactory#getNotificationService returning old Servic
     expect(result).toEqual(oldNotificationService);
   });
 
-  it("should return new notification service if ff is CANARY and user is not a beta tester", async () => {
+  it("should return old notification service if ff is CANARY and user is not a beta tester neither he bolongs to canary set", async () => {
```
Comment (AleDore) [inline src/services/__tests__/notificationServiceFactory.test.ts:49]:
```suggestion
  it("should return old notification service if ff is CANARY and user is not a beta tester neither he belongs to canary set", async () => {
```

**Review by fabriziopapi — APPROVED** [review]

**Pull request merged**